### PR TITLE
[16.0][FIX] product_tax_multicompany_default: Improve test coverage

### PR DESCRIPTION
**Pull request by Shide**
Testing `product.set_multicompany_taxes()` on `product_tax_multicompany_default ` module

Cherry pick of https://github.com/OCA/multi-company/pull/447/commits/9645c0c54c5cc45649d274fde607155e2cb998c9

MT-2121 @moduon @rafaelbn @carlosdauden @yajo

### CODE REVIEW

**Review by yajo — APPROVED** [review]

**Review by carlosdauden — APPROVED** [review]

File: product_tax_multicompany_default/tests/test_product_tax_multicompany.py [hunk 1]
```diff
@@ -17,27 +19,49 @@ def setUpClass(cls):
             {"name": "Test company 2", "country_id": default_country.id}
         )
         group_account_manager = cls.env.ref("account.group_account_manager")
+        group_account_user = cls.env.ref("account.group_account_user")
+        group_multi_company = cls.env.ref("base.group_multi_company")
         ResUsers = cls.env["res.users"]
         cls.user_1 = ResUsers.create(
```
Comment (pedrobaeza) [inline product_tax_multicompany_default/tests/test_product_tax_multicompany.py]: Please use the method `test_new_user` for simplifying code:

https://github.com/odoo/odoo/blob/08859ca7939454303e34dc39c831ac8f75773c3d/odoo/tests/common.py#L124

Example: https://github.com/odoo/odoo/blob/08859ca7939454303e34dc39c831ac8f75773c3d/addons/web/tests/test_ir_model.py#L37

File: product_tax_multicompany_default/tests/test_product_tax_multicompany.py [hunk 2]
```diff
@@ -131,3 +167,75 @@ def test_tax_not_default_set_match(self):
         product = product.sudo()
         self.assertIn(self.tax_10_cc2, product.taxes_id)
         self.assertIn(self.tax_10_sc2, product.supplier_taxes_id)
+
+    def test_set_multicompany_taxes(self):
```
Comment (pedrobaeza) [inline product_tax_multicompany_default/tests/test_product_tax_multicompany.py:141]: Why not decorating this with `@users` for testing with such user/s?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Please put the commit message properly

File: product_tax_multicompany_default/tests/test_product_tax_multicompany.py (hunk 2 above)
Comment (Shide) [reply product_tax_multicompany_default/tests/test_product_tax_multicompany.py:141]: Done

File: product_tax_multicompany_default/tests/test_product_tax_multicompany.py (hunk 1 above)
Comment (Shide) [reply product_tax_multicompany_default/tests/test_product_tax_multicompany.py]: Done

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge patch

**Pull request merged**